File: backend/HerdFlow.Api/Models/Cow.cs
using Microsoft.EntityFrameworkCore;

namespace HerdFlow.Api.Models;

[Index(nameof(TagNumber), IsUnique = true)]
public class Cow
{
    public int Id { get; set; }

    public string TagNumber { get; set; } = null!;
    public string Breed { get; set; } = null!;
    public string HealthStatus { get; set; } = null!;
    public string HeatStatus { get; set; } = null!;
    public string BreedingStatus { get; set; } = null!;

    public string? OwnerName { get; set; }
    public DateOnly? DateOfBirth { get; set; }
    public decimal? PurchasePrice { get; set; }
    public decimal? SalePrice { get; set; }
}

File: frontend/src/types/cow.ts
export type Cow = {
  id: number;
  tagNumber: string;
  breed: string;
  healthStatus: "Healthy" | "NeedsTreatment";
  heatStatus: string;
  breedingStatus: string;
  ownerName?: string | null;
  dateOfBirth?: string | null;
  purchasePrice?: number | null;
  salePrice?: number | null;
};
